File: Data Structures & Algorithms/two-integer-sum/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums: number[], target: number): number[] {
        let i = 0; // left, slow
        let j = 1; // right, fast
        let sum = 0;

        while(i < nums.length -1 ){
            sum = nums[i] + nums[j];
            if(sum === target){
                 return [i, j]
            }
            if(j < nums.length - 1){
                j++;
            }else{
                i++;
                j = i + 1;
            }

        
        }

        return [];
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-0.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        Integer length = nums.length;
        int[] ans = new int[length * 2];
        
        for(Integer i = 0; i < length; i++){
            ans[i] = nums[i];
            ans[i + length] = nums[i];
        }

        return ans;
    }
}